src/rdkafka_msg.c

```diff
@@ -570,6 +541,35 @@ int32_t rd_kafka_msg_partitioner_consistent_random (const rd_kafka_topic_t *rkt,
                                                  msg_opaque);
 }
 
+int32_t rd_kafka_msg_partitioner_murmur2_consistent (const rd_kafka_topic_t *rkt,
+ 					     const void *key, size_t keylen,
+					     int32_t partition_cnt,
+					     void *rkt_opaque,
+					     void *msg_opaque) {
+    return rd_murmur2(key, keylen) % partition_cnt;
+}
+
+int32_t rd_kafka_msg_partitioner_murmur2_random (const rd_kafka_topic_t *rkt,
+					 const void *key, size_t keylen,
+					 int32_t partition_cnt,
+					 void *rkt_opaque,
+					 void *msg_opaque) {
+	if (keylen == 0)
```

Comment:
Looking at the Java partitioner I believe this should probably be `if (!key)` since there is a difference between no key (null) and an empty key (which is just an arbitrary key).
If so this is also incorrect on the existing partitioners, but we can't change their behaviour.